File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/filter/FilterParser.java
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.filter;

import io.github.oleksivio.telegram.bot.core.controller.annotations.parser.Parser;
import io.github.oleksivio.telegram.bot.core.controller.annotations.parser.finder.Finder;
import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator;

import java.lang.annotation.Annotation;

public interface FilterParser<A extends Annotation, IN> extends Parser<A> {

    Validator<IN> createChecker(A annotation,
                                Finder finder);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/controller/annotations/parser/filter/FilterParser.kt
package io.github.oleksivio.tl.kbot.core.controller.annotations.parser.filter

import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.Parser
import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.finder.Finder
import io.github.oleksivio.tl.kbot.core.controller.handler.check.Validator

interface FilterParser<A : Annotation, IN> :
    Parser<A> {

    fun createChecker(annotation: A, finder: Finder): Validator<IN>
}
